core/src/consensus/stake/action_data.rs

```diff
@@ -1129,46 +1182,85 @@ mod tests {
-        let mut origin_rewards = IntermediateRewards::load_from_state(&state).unwrap();
+        let mut origin_rewards = IntermediateRewardsV0::load_from_state(&state).unwrap();
         origin_rewards.add_quantity(address1, 1);
         origin_rewards.add_quantity(address2, 2);
         origin_rewards.save_to_state(&mut state).unwrap();
-        let mut recovered_rewards = IntermediateRewards::load_from_state(&state).unwrap();
+        let mut recovered_rewards = IntermediateRewardsV0::load_from_state(&state).unwrap();
         assert_eq!(origin_rewards, recovered_rewards);
         let _drained = recovered_rewards.drain_previous();
         recovered_rewards.save_to_state(&mut state).unwrap();
-        let mut final_rewards = IntermediateRewards::load_from_state(&state).unwrap();
+        let mut final_rewards = IntermediateRewardsV0::load_from_state(&state).unwrap();
         assert_eq!(BTreeMap::new(), final_rewards.previous);
         let current = final_rewards.current.clone();
         final_rewards.move_current_to_previous();
         assert_eq!(BTreeMap::new(), final_rewards.current);
         assert_eq!(current, final_rewards.previous);
     }
 
+    #[test]
+    fn load_and_save_intermediate_rewards_v1() {
+        let mut state = helpers::get_temp_state();
+        let rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        rewards.save_to_state(&mut state).unwrap();
+    }
+
+    #[test]
+    fn add_quantity_v1() {
+        let address1 = Address::random();
+        let address2 = Address::random();
+        let mut state = helpers::get_temp_state();
+        let mut origin_rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        origin_rewards.add_quantity(address1, 1);
+        origin_rewards.add_quantity(address2, 2);
+        origin_rewards.save_to_state(&mut state).unwrap();
+        let recovered_rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        assert_eq!(origin_rewards, recovered_rewards);
+    }
+
+    #[test]
+    fn drain_v1() {
+        let address1 = Address::random();
+        let address2 = Address::random();
+        let mut state = helpers::get_temp_state();
+        let mut origin_rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        origin_rewards.add_quantity(address1, 1);
+        origin_rewards.add_quantity(address2, 2);
+        origin_rewards.save_to_state(&mut state).unwrap();
+        let mut recovered_rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        assert_eq!(origin_rewards, recovered_rewards);
+        recovered_rewards.drain_current();
+        recovered_rewards.save_to_state(&mut state).unwrap();
+        let mut final_rewards = IntermediateRewardsV1::load_from_state(&state).unwrap();
+        assert_eq!(BTreeMap::new(), final_rewards.current);
+        final_rewards.drain_calculated();
+        assert_eq!(BTreeMap::new(), final_rewards.calculated);
+    }
+
     #[test]
```

Comment:
Can you add a test save in V0 -> load V1?

Reply:
Fixed. It required some `pub(super)` modifiers, so the code became a bit ugly :(